x-pack/plugins/stack_connectors/public/connector_types/slack_api/slack_params.tsx

```diff
@@ -23,17 +35,112 @@ const SlackParamsFields: React.FunctionComponent<ActionParamsProps<PostMessagePa
   defaultMessage,
   useDefaultMessage,
 }) => {
+  const [connectorId, setConnectorId] = useState<string>();
   const { subAction, subActionParams } = actionParams;
-  const { channels = [], text } = subActionParams ?? {};
+  const { channels = [], text, channelIds = [] } = subActionParams ?? {};
+  const [tempChannelId, setTempChannelId] = useState(
+    channels.length > 0
+      ? channels[0]
+      : channelIds.length > 0 && channelIds[0].length > 0
+      ? channelIds[0]
+      : ''
+  );
+  const [validChannelId, setValidChannelId] = useState('');
   const { toasts } = useKibana().notifications;
+  const [selectedChannels, setSelectedChannels] = useState<EuiComboBoxOptionOption[]>(
+    (channelIds ?? []).map((c) => {
+      const allowedChannels = (
+        actionConnector as UserConfiguredActionConnector<SlackApiConfig, unknown>
+      )?.config?.allowedChannels;
+      const allowedChannelSelected = allowedChannels?.find((ac) => ac.id === c);
+      return {
+        value: c,
+        label: allowedChannelSelected
+          ? `${allowedChannelSelected.id} - ${allowedChannelSelected.name}`
+          : c,
+      };
+    })
+  );
+  const [channelValidError, setChannelValidError] = useState<string[]>([]);
+  const {
+    response: { channel: channelValidInfo } = {},
+    isLoading: isValidatingChannel,
+    error: channelValidErrorResp,
+  } = useSubAction<ValidChannelIdSubActionParams, ValidChannelResponse>({
+    connectorId: actionConnector?.id,
+    subAction: 'validChannelId',
+    subActionParams: {
+      channelId: validChannelId,
+    },
+    disabled: validChannelId.length === 0,
+  });
 
   useEffect(() => {
     if (useDefaultMessage || !text) {
-      editAction('subActionParams', { channels, text: defaultMessage }, index);
+      editAction('subActionParams', { channels, channelIds, text: defaultMessage }, index);
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [defaultMessage, useDefaultMessage]);
 
+  useEffect(() => {
+    if (
+      !isValidatingChannel &&
+      !channelValidErrorResp &&
+      channelValidInfo &&
+      validChannelId === channelValidInfo.id
+    ) {
+      editAction(
+        'subActionParams',
+        { channels: undefined, channelIds: [channelValidInfo.id], text },
+        index
+      );
+      setValidChannelId('');
+      setChannelValidError([]);
+    }
+  }, [
+    channelValidInfo,
+    validChannelId,
+    channelValidErrorResp,
+    isValidatingChannel,
+    editAction,
+    text,
+    index,
+  ]);
+
+  useEffect(() => {
+    if (channelValidErrorResp && validChannelId.length > 0) {
+      editAction('subActionParams', { channels: undefined, channelIds: [], text }, index);
+      const errorMessage = i18n.translate(
+        'xpack.stackConnectors.slack.params.componentError.validChannelsRequestFailed',
+        {
+          defaultMessage: '{validChannelId} is not a valid Slack channel',
```

Comment:
>          defaultMessage: '{validChannelId} is not a valid Slack channel',

Is it not a valid Slack channel (i.e. doesn't exist in Slack) or not an allowed channel (i.e. doesn't exist in the connector's list of allowed channels)?